osquery/config/config.cpp

```diff
@@ -455,6 +468,16 @@ Status Config::refresh() {
     }
 
     loaded_ = true;
+
+    if (FLAGS_config_enable_backup) {
```

Comment:
This doesn't need to be run every time a refresh fails. In fact it doesn't make much sense since you'll basically replace the config with the already existing config. This should run only once when osquery is starting.

Reply:
I don't want to update it when osquery starts. I think, it's better to give config plugin chance to 'naturally refresh' and if it fails only then to restore the config. Backed up config can be very old or something, also why should we force osquery to get config faster than it was intended to(plugin was configured to update later maybe).

If you don't mind, will just make sure that it happens only once in the refresh.